tests/sources/test_outlook.py

```diff
@@ -597,13 +624,49 @@ async def test_get_users_for_cloud():
         ):
             with mock.patch(
                 "aiohttp.ClientSession.get",
-                side_effect=side_effect_function,
+                side_effect=side_effect_function(),
             ):
                 async for response in source.client._get_user_instance.get_users():
                     user_mails = [user["mail"] for user in response["value"]]
                     users.extend(user_mails)
                 assert users == ["test.user@gmail.com", "dummy.user@gmail.com"]
 
+    client_emails = "one.user@gmail.com"
+    async with create_outlook_source(client_emails=client_emails) as source:
```

Comment:
I suggest making these their own test functions. That way, any failures are more easily identifyable, and they can have more focused test names to help the reader understand what's being tested.

Reply:
I've added a test function to cover the batch POST request handling for fetching user data with client emails